### DISCUSSION TITLE
how could i start gulp tasks programmatically?

### DISCUSSION THREAD

**Discussion by zzzgit** [gulpjs/gulp#2672] [Help] [answered]
We can invoke gulp use `gulp-cli`, we can invoke it via an IDE plugin.

However, can I invoke it programmatically? like: 

`gulp.start()`
or 
`gulp.run()`

**Comment by phated** [comment, marked as answer]
Tasks are just functions. If you do this:
```js
const task = parallel(foo, bar);
console.log(task);
```
You'll see it is a function. You can just call it!
```js
const task = parallel(foo, bar);
task()
```

These are referred to as "private tasks" in https://gulpjs.com/docs/en/getting-started/creating-tasks 
> A private task looks and acts like any other task, but an end-user can't ever execute it independently. To register a task publicly, export it from your gulpfile.
